sqlalchemy_utils/functions/database.py

```diff
@@ -442,11 +445,8 @@ def database_exists(url):
     """
 
     def get_scalar_result(engine, sql):
-        result_proxy = engine.execute(sql)
-        result = result_proxy.scalar()
-        result_proxy.close()
-        engine.dispose()
-        return result
+        with engine.connect() as conn:
+            return conn.scalar(sql)
```

Comment:
Since it's only a two-liner, is the inline function still needed?

Reply:
I would say so. Still avoids some code duplication. Do you have some alternative in mind?

Reply:
That's true, but personally I don't really like dynamically defined functions. Maybe move it to the module level?

Reply:
> personally I don't really like dynamically defined functions

Can do. Out of interest: is there a downside, apart from violating style preferences?

Reply:
The function is created every time and that most likely eats up a bit of performance.